scripts/mop_workspace/mop_workspace.py

```diff
@@ -0,0 +1,474 @@
+import argparse
+import os
+
+from concurrent.futures import ThreadPoolExecutor
+from datetime import timedelta, datetime
+from firecloud import api as fapi
+from google.cloud import storage  # note: please use version 2.5.0
+from io import open
+from six import string_types
+from toolz.itertoolz import partition_all
+from tqdm import tqdm
+from time import sleep
+
+
+def _entity_paginator(namespace, workspace, etype, page_size=500,
+                      filter_terms=None, sort_direction="asc"):
+    """Pages through the get_entities_query endpoint to get all entities in
+       the workspace without crashing.
+    """
+
+    page = 1
+    all_entities = []
+    # Make initial request
+
+    def get_page():
```

Comment:
in my experience it's usually best practice not to define functions inside other functions, unless there's a good reason for it. 

defining this here means the function will be re-defined every time the parent `_entity_paginator()` function is called. I'd suggest moving `get_page()` out of this function (you'll have to define the input parameters you need) and probably rename it to `_get_page()` to denote it as "private"

see https://realpython.com/inner-functions-what-are-they-good-for/ : "Although writing your helper functions as inner functions achieves the desired result, you’ll probably be better served by extracting them as top-level functions. In this case, you could use a leading underscore (_) in the name of the function to indicate that it’s private to the current [module](https://realpython.com/python-modules-packages/) or class. This will allow you to access your helper functions from anywhere else in the current module or class and reuse them as needed."